src/armature/GuidingVectors.ts

```diff
@@ -219,6 +237,36 @@ export class GuidingVectors implements CostFn {
         return { realCost: totalCost, heuristicCost };
     }
 
+    /**
+     * Returns an array representing whether or not each guiding curve has been the
+     * closest one to a bone yet.
+     *
+     * @param {GeneratorInstance} instance The instance that we are checking.
+     * @param {Node[]} added The nodes that were added to this instance since the last round.
+     * @returns {boolean[]} An array representing whether guide i has been closest to any
+     * bone yet.
+     */
+    private getOrCreateAnyClosest(instance: GeneratorInstance, added: Node[]): boolean[] {
+        const lastNode = instance.getModel().nodes[instance.getModel().nodes.length - 1];
```

Comment:
Nitpick: you use this expression `instance.getModel().nodes.length - 1` a few times in the function. Could you create a local variable for it, please?